flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/LegacyScheduler.java

```diff
@@ -526,11 +526,13 @@ public void acknowledgeCheckpoint(final JobID jobID, final ExecutionAttemptID ex
 			checkpointId,
 			checkpointMetrics,
 			checkpointState);
+		final Execution currentExecution = executionGraph.getRegisteredExecutions().get(executionAttemptID);
```

Comment:
refactor into method `Optional<TaskManagerLocation> tryRetrieveTaskManagerLocation(ExecutionAttemptID)` and reuse in `declineCheckpoint`